coalib/io/FileProxy.py

```diff
@@ -0,0 +1,363 @@
+import logging
+from os import path
+
+from coala_utils.FileUtils import detect_encoding
+from coala_utils.decorators import (enforce_signature, generate_eq,)
+
+
+class FileNameError(LookupError):
+    def __init__(self, error_message):
```

Comment:
you sure you need this?

it isnt doing anything afaics

Reply:
I wanted to have a distinct error to denote bad file names. I will use `ValueError` instead.